torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -290,19 +290,24 @@ struct PythonPrintImpl {
     }
   }
 
-  size_t getOrAddTensorConstant(at::Tensor t) {
-    // XXX - N^2 warning. This code does the exact same thing as
-    // ConstantPool, which is also N^2 in the size of the constants,
-    // because it doesn't hash any information about the tensors.
-    // We will probably need to optimize this at some point using hashing.
-    for (size_t i = 0; i < tensor_table_.size(); ++i) {
-      if (t.options().type_equal(tensor_table_[i].options()) &&
-          t.equal(tensor_table_[i])) {
-        return i;
+  size_t getOrAddTensorConstant(at::IValue val) {
+    if (val.isTensor()) {
+      auto t = val.toTensor();
+      // XXX - N^2 warning. This code does the exact same thing as
+      // ConstantPool, which is also N^2 in the size of the constants,
+      // because it doesn't hash any information about the tensors.
+      // We will probably need to optimize this at some point using hashing.
+      for (size_t i = 0; i < constant_table_.size(); ++i) {
+        if (!constant_table_[i].isTensor())
+          continue;
+        auto t2 = constant_table_[i].toTensor();
+        if (t.options().type_equal(t2.options()) && t.equal(t2)) {
+          return i;
+        }
       }
     }
-    tensor_table_.emplace_back(std::move(t));
-    return tensor_table_.size() - 1;
+    constant_table_.emplace_back(std::move(val));
```

Comment:
Looks like we're now only pooling tensors but not other constants. I do wonder—why are we pooling tensors at all? Shouldn't constant pooling do it for us? Maybe we should remove this manual pooling code altogether.

Reply:
I agree! I was going to disable pooling.

Reply:
The pooling is out scope. I can do it in a separate patch. There are some tests that fails that need to be fixed